payjoin-ffi/dart/test/test_payjoin_integration_test.dart

```diff
@@ -0,0 +1,407 @@
+import "dart:convert";
+import "dart:typed_data";
+
+import "package:http/http.dart" as http;
+import 'package:test/test.dart';
+import "package:convert/convert.dart";
+
+import "../lib/payjoin_ffi.dart" as payjoin;
+import "../lib/bitcoin.dart" as bitcoin;
+
+late payjoin.BitcoindEnv env;
+late payjoin.BitcoindInstance bitcoind;
+late payjoin.RpcClient receiver;
+late payjoin.RpcClient sender;
+
+class InMemoryReceiverPersister
+    implements payjoin.JsonReceiverSessionPersister {
+  final String id;
+  final List<String> events = [];
+  bool closed = false;
+
+  InMemoryReceiverPersister(this.id);
+
+  @override
+  void save(String event) {
+    events.add(event);
+  }
+
+  @override
+  List<String> load() {
+    return events;
+  }
+
+  @override
+  void close() {
+    closed = true;
+  }
+}
+
+class InMemorySenderPersister implements payjoin.JsonSenderSessionPersister {
+  final String id;
+  final List<String> events = [];
+  bool closed = false;
+
+  InMemorySenderPersister(this.id);
+
+  @override
+  void save(String event) {
+    events.add(event);
+  }
+
+  @override
+  List<String> load() {
+    return events;
+  }
+
+  @override
+  void close() {
+    closed = true;
+  }
+}
+
+class MempoolAcceptanceCallback implements payjoin.CanBroadcast {
+  final payjoin.RpcClient connection;
+
+  MempoolAcceptanceCallback(this.connection);
+
+  @override
+  bool callback(Uint8List tx) {
+    try {
+      final hexTx = bytesToHex(tx);
+      final resultJson = connection.call("testmempoolaccept", ['["$hexTx"]']);
+      final decoded = jsonDecode(resultJson);
+      return decoded[0]['allowed'] == true;
+    } catch (e) {
+      print("An error occurred: $e");
+      return false;
+    }
+  }
+
+  String bytesToHex(Uint8List bytes) {
+    return bytes.map((b) => b.toRadixString(16).padLeft(2, '0')).join();
+  }
+}
+
+class IsScriptOwnedCallback implements payjoin.IsScriptOwned {
+  final payjoin.RpcClient connection;
+
+  IsScriptOwnedCallback(this.connection);
+
+  @override
+  bool callback(Uint8List script) {
+    try {
+      final scriptObj = bitcoin.Script(script);
+      final address =
+          bitcoin.Address.fromScript(scriptObj, bitcoin.Network.regtest);
+      // This is a hack due to toString() not being exposed by dart FFI
+      final address_str = address.toQrUri().split(":")[1];
+      final result = connection.call("getaddressinfo", [address_str]);
+      final decoded = jsonDecode(result);
+      return decoded["ismine"] == true;
+    } catch (e) {
+      print("An error occurred: $e");
+      return false;
+    }
+  }
+}
+
+class CheckInputsNotSeenCallback implements payjoin.IsOutputKnown {
```

Comment:
Something seems a bit funky here (?) Should this class be `CheckInOutputKnownCallback`

Reply:
For inputs seen before I dont think there is much to implement here but  for output known could we use the same functionality as `IsScriptOwned`? 

Edit: that actually might be what we already do below.

Reply:
Nevermind we do this in the python int. tests as well. Appears like the FFI exported abstract class is misnamed?
```python
class CheckInputsNotSeenCallback(IsOutputKnown):
    def __init__(self, connection: RpcClient):
        self.connection = connection

    def callback(self, _outpoint):
        return False
```